colobot-base/src/common/system/system.cpp

```diff
@@ -149,13 +146,7 @@ SystemDialogResult CSystemUtils::ConsoleSystemDialog(SystemDialogType type, cons
 
 std::filesystem::path CSystemUtils::GetBasePath()
 {
-    if (m_basePath.empty())
-    {
-        auto* path = SDL_GetBasePath();
-        m_basePath = StrUtils::ToPath(path);
-        SDL_free(path);
-    }
-    return m_basePath;
+    return std::filesystem::current_path();
```

Comment:
Why? Before this `PORTABLE` builds searched for files relative to the location of the executable. Now we will only find the files if the current working directory happens to be the directory of the executable. Isn't this strictly less reliable?

Reply:
Less reliable how? Unless you specifically change it, the current directory is the same as directory of the executable so there's no difference in behavior. This change removes requirement for strict location of the executable in relation to game files. You will now be able to place it in `bin` subdirectory, for example. In some ways it's more portable than before.

This change also allows running the project from the build directory on multi-config systems, like MSVC. There, the executable is placed in config-specific subdirectory even if you point `CMAKE_RUNTIME_OUTPUT_DIRECTORY` to the base build directory, meaning that running the project after building is impossible without first copying the executable. With this change you can run the executable from any build location because you can manually point the current directory to the base build directory and the game will find its assets normally.

Unless you mean that the current directory might be changed during execution and corrupt something as a result because the returned value is different. In that case we can cache the current directory on creation so even if it changes this method will return the same value.

Reply:
Actually, the current directory is already cached as `m_basePath` in the constructor if you check the updated code:
```
CSystemUtils::CSystemUtils()
{
    m_basePath = std::filesystem::current_path();
}
```
The returned base directory value cannot change on its own. The value can still be changed by a derived class and this is done in the Linux implementation when building for AppImage is enabled. I think this is fine.

Reply:
> Unless you specifically change it, the current directory is the same as directory of the executable

Not if stared from the command line.

Reply:
> Not if stared from the command line.

Running an executable from a directory other than its own directory is a deliberate choice and not something you do randomly. Under normal circumstances you don't want to run the game from any other directory, even from terminal.

And as a developer you want to be able to change executable's behavior by using whatever current directory is convenient to you. In this case, for the purpose of running the game from the build directory. I do this all the time as a developer in most of my projects.

Also, AppImage has override to use old behavior. Do you think there should be a CMake option for all platforms to change whether base path is set to executable's directory or current directory?

Reply:
> Do you think there should be a CMake option for all platforms to change whether base path is set to executable's directory or current directory?

No. No. No. There are too many build options as is. It's fine